sensor/common/registry/registry_store.go

```diff
@@ -392,3 +392,221 @@ func (rs *Store) GetMatchingCentralRegistryIntegrations(imgName *storage.ImageNa
 
 	return regs
 }
+
+// UpsertSecret upserts a pull secret into the store.
+func (rs *Store) UpsertSecret(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	if !features.SensorPullSecretsByName.Enabled() {
+		rs.upsertSecretByHost(namespace, secretName, dockerConfig, serviceAcctName)
+		return
+	}
+
+	rs.upsertSecretByName(namespace, secretName, dockerConfig, serviceAcctName)
+}
+
+func (rs *Store) upsertSecretByHost(namespace, secretName string, dockerConfig config.DockerConfig, serviceAcctName string) {
+	isGlobalPullSecret := namespace == openshiftConfigNamespace && secretName == openshiftConfigPullSecret
+	fromDefaultSA := serviceAcctName == defaultSA
```

Comment:
There used to be a comment associated with this line. Can we add it back? I helped explain why this was useful

Reply:
Sure, I felt that the comment a few lines later explains the what/why, will add the original comment back here:

> 	// In Kubernetes, the `default` service account always exists in each namespace (it is recreated upon deletion).
	// The default service account always contains an API token.
	// In OpenShift, the default service account also contains credentials for the
	// OpenShift Container Registry, which is an internal image registry.